src/workerd/api/util-test.c++

```diff
@@ -53,5 +63,83 @@ KJ_TEST("redactUrl can detect base64 ids") {
   expectUnredacted("https://domain/IThinkIShallNeverSee0/x"_kj);
 }
 
+KJ_TEST("readContentTypeParameter can fetch boundary parameter") {
+
+  // normal
+  expectContentTypeParameter(
+    "multipart/form-data; boundary=\"__boundary__\""_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // multiple params
+  expectContentTypeParameter(
+    "multipart/form-data; charset=utf-8; boundary=\"__boundary__\""_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+ // param name inside value of other param
+  expectContentTypeParameter(
+    "multipart/form-data; charset=\"boundary=;\"; boundary=\"__boundary__\""_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // ensure param is not found
+  KJ_ASSERT(readContentTypeParameter(
+    "multipart/form-data; charset=\"boundary=;\"; boundary=\"__boundary__\""_kj,
+    "boundary1"_kj
+  ) == nullptr);
+
+  // no quotes
+  expectContentTypeParameter(
+    "multipart/form-data; charset=\"boundary=;\"; boundary=__boundary__"_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // attribute names are case-insensitive, but values are not
+  expectContentTypeParameter(
+    "multipart/form-data; charset=\"boundary=;\"; boundary=__Boundary__"_kj,
+    "Boundary"_kj,
+    "__Boundary__"_kj
+  );
+
+  // different order
+  expectContentTypeParameter(
+    "multipart/form-data; boundary=\"__boundary__\"; charset=utf-8"_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // bogus parameter
+  expectContentTypeParameter(
+    "multipart/form-data; foo=123; boundary=\"__boundary__\""_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // quoted-string
+  expectContentTypeParameter(
+    "multipart/form-data; foo=\"\\\"boundary=bar\\\"\"; boundary=\"realboundary\""_kj,
```

Comment:
No need to change anything, but I wanted to mention that raw string literals might make these strings easier to write, e.g.
```cpp
R"(multipart/form-data; foo="\"boundary=bar\""; boundary="realboundary")",
```

Reply:
Changed.